salt/roster/terraform.py

```diff
@@ -0,0 +1,237 @@
+    infrastructure, so keep it safe. To inspect the complete state
+    use the `terraform show` command.
+
+    State path: terraform.tfstate
+
+You can start now using salt-ssh without any configuration.
+
+.. code-block:: bash
+
+   $ salt-ssh '*' test.ping
+   domain-0:
+       True
+   domain-1:
+       True
+
+'''
+from __future__ import absolute_import
+import logging
+import json
+import os.path
+import salt.utils
+from salt.roster.flat import RosterMatcher
+# pylint: disable=import-error
+from salt.ext import six
+# pylint: disable=redefined-builtin
+from salt.ext.six.moves import range
+# pylint: enable=import-error,redefined-builtin
+
+log = logging.getLogger(__name__)
+
+
+def _handle_libvirt_resource(resource, ret):
+    '''
+    Handles libvirt resources.
+    See https://github.com/dmacvicar/terraform-provider-libvirt
+    '''
+    primary = resource['primary']
+    attrs = primary['attributes']
+    name = attrs['name']
+    num_ifaces = int(attrs['network_interface.#'])
+    for if_k in range(num_ifaces):
+        num_addrs = int(attrs['network_interface.{}.addresses.#'.format(if_k)])
+        for addr_k in range(num_addrs):
+            addr = attrs['network_interface.{}.addresses.{}'.format(if_k, addr_k)]
+            if not addr:
+                continue
+            ret[name] = {'host': addr}
+            _add_ssh_key(ret[name])
+            return
+
+
+def _handle_aws_resource(resource, ret):
+    '''
+    Reads public_dns or public_ip attributes from AWS resources.
+    '''
+    primary = resource['primary']
+    attrs = primary['attributes']
+
+    addr = attrs['public_dns']
+    if not addr:
+        addr = attrs['public_ip']
+    if not addr:
+        return
+
+    name = attrs['tags.Name']
+    if not name:
+        name = attrs['id']
+    if not name:
+        return
+
+    ret[name] = {'host': addr}
+    _add_ssh_key(ret[name])
+
+
+def _add_ssh_key(ret):
+    '''
+    Setups the salt-ssh minion to be accessed with salt-ssh default key
+    '''
+    priv = None
+    if __opts__.get('ssh_use_home_key') and \
+       os.path.isfile(os.path.expanduser('~/.ssh/id_rsa')):
+        priv = os.path.expanduser('~/.ssh/id_rsa')
+    else:
+        priv = __opts__.get(
+            'ssh_priv',
+            os.path.abspath(os.path.join(
+                __opts__['pki_dir'],
+                'ssh',
+                'salt-ssh.rsa'
+            ))
+        )
+    if not priv:
+        return
+    if os.path.isfile(priv):
+        ret['priv'] = priv
+
+
+def _parse_state_file(state_file_path='terraform.tfstate'):
+    '''
+    Parses the terraform state file passing different resource types to the right handler
+    '''
+    ret = {}
+    with salt.utils.fopen(state_file_path, 'r') as fh_:
+        tfstate = json.load(fh_)
+
+    modules = tfstate.get('modules', None)
+
+    if not modules:
+        log.error('Malformed tfstate file. No modules found')
+        return ret
```

Comment:
I would not make `modules` as `None` for an extra `return` in the middle, but make it iterable (`dict` in your case). So the loop below will skip it.